airbyte-cdk/bulk/toolkits/load-dlq/src/main/kotlin/io/airbyte/cdk/load/command/dlq/ObjectStorageConfig.kt

```diff
@@ -0,0 +1,82 @@
+import io.airbyte.cdk.load.command.object_storage.ObjectStoragePathConfigurationProvider
+import io.airbyte.cdk.load.command.s3.S3BucketConfiguration
+import io.airbyte.cdk.load.command.s3.S3BucketConfigurationProvider
+import java.io.OutputStream
+
+interface ObjectStorageConfigProvider {
+    val objectStorageConfig: ObjectStorageConfig
+}
+
+sealed interface ObjectStorageConfig {
+    val type: String
+}
+
+class DisabledObjectStorageConfig : ObjectStorageConfig {
+    override val type: String = "None"
+}
+
+class S3ObjectStorageConfig<T : OutputStream>(
+    override val awsAccessKeyConfiguration: AWSAccessKeyConfiguration,
+    override val awsArnRoleConfiguration: AWSArnRoleConfiguration,
+    override val s3BucketConfiguration: S3BucketConfiguration,
+    override val objectStorageCompressionConfiguration: ObjectStorageCompressionConfiguration<T>,
+    override val objectStorageFormatConfiguration: ObjectStorageFormatConfiguration,
+    override val objectStoragePathConfiguration: ObjectStoragePathConfiguration,
+) :
+    ObjectStorageConfig,
+    AWSAccessKeyConfigurationProvider,
+    AWSArnRoleConfigurationProvider,
+    S3BucketConfigurationProvider,
+    ObjectStorageCompressionConfigurationProvider<T>,
+    ObjectStorageFormatConfigurationProvider,
+    ObjectStoragePathConfigurationProvider {
+    override val type: String = "S3"
+}
+
+fun ObjectStorageSpec.toObjectStorageConfig(): ObjectStorageConfig =
+    when (this) {
+        is DisabledObjectStorageSpec -> DisabledObjectStorageConfig()
+        is S3ObjectStorageSpec ->
+            S3ObjectStorageConfig(
+                awsAccessKeyConfiguration = toAWSAccessKeyConfiguration(),
+                awsArnRoleConfiguration = toAWSArnRoleConfiguration(),
+                s3BucketConfiguration = toS3BucketConfiguration(),
+                objectStorageCompressionConfiguration = toCompressionConfiguration(),
+                objectStorageFormatConfiguration = toObjectStorageFormatConfiguration(),
+                objectStoragePathConfiguration =
+                    ObjectStoragePathConfiguration(
+                        prefix = bucketPath,
+                        pathPattern = normalizePathFormat(pathFormat),
+                        fileNamePattern = fileNameFormat,
+                    )
+            )
+    }
+
+/**
+ * Normalize the path format to match how the internals of the object storage toolkit work.
+ *
+ * - pathPattern requires variables to be `${VAR}` (both $ and Uppercase)
+ * - pathPattern doesn't guarantee it is only a path, to avoid path leaking into the filename, we
+ * add a trailing '/'.
+ */
+internal fun normalizePathFormat(format: String?): String? =
+    format?.let {
+        val pathWithCorrectVars =
+            it.replace("""\{\w+}""".toRegex()) { match -> "${'$'}${match.value.uppercase()}" }
+        if (pathWithCorrectVars.endsWith('/')) pathWithCorrectVars else "$pathWithCorrectVars/"
+    }
```

Comment:
Does this mean users won't have to use two different interpolation formats?

Reply:
Yes, this is to align both format when using this new configuration flow. A quick fix to stop the inconsistency without breaking existing connectors.